### ISSUE TITLE
Run on newer tensorfolow

### ISSUE THREAD

**Issue by yerzhanimanmalik** [pbrakel/anica#2]
Great work! 
It seems to be applicable to the research that I am currently doing.

However, I am struggling to run this code on Tensorflow2 and Python3.
Do you aware of the ways I can do it? Or do you have any other suggestions?
Perhaps some other pytorch implementations of adversarial nonlinear ica are existent

Thanks in advance

**Comment by pbrakel** [comment]
Hi,

Thanks for your interest and reaching out.

I haven't maintained this code in a while but if you have Tensorflow2 installed, replacing occurrences of `import tensorflow as tf` with `import tensorflow.compat.v1 as tf` would be the first thing to do and it might be sufficient but see https://www.tensorflow.org/guide/upgrade for more info. Python3 compatibility is a bit harder to predict. I might add this in the future but if you get it to work yourself, a pull request would be very welcome :)

I don't know of any other implementations of exactly this algorithm (it's research code) but the factorized VAE is almost the same idea (although not used for ICA) and can be found here: https://github.com/YannDubs/disentangling-vae. You might also want to check out Devon Hjelm's work on Deep Infomax: https://github.com/rdevon/DIM

Best,

Philemon

**Comment by yerzhanimanmalik** [comment]
Thank you for your answer!

Yeah sure, if I'll manage to run it or going to implement it on pytorch, I'll let you know :)

Thanks for pointing out the works on factorized VAE and Deep Infomax. I am looking for the ways to do fMRI component analysis using neural nets instead of ICA. If by any chance you have encountered any other use of deep learning for ICA, I'd be very grateful if you share the links.

Sincerely, Yerzhan

**Comment by yerzhanimanmalik** [comment]
Philemon, thanks again for your previous answer!

I found this recent work on non-linear ICA, just wanted to share with you [https://github.com/ilkhem/icebeem]

**Comment by pbrakel** [comment]
Hi Yerzhan,

Thanks for sharing that paper! It looks interesting.

Best,

Philemon
